rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportLogger.scala

```diff
@@ -93,114 +96,166 @@ class AutomaticReportLogger(
+            logger.warn("Automatic report logger has never run, logging latest 100 non compliant reports")
+            val isSuccess = for {
+              hundredReports <- reportsRepository.getLastHundredErrorReports(reportskind)
+              nodes          <- nodeInfoService.getAll
+              rules          <- ruleRepository.getAll(true)
+              directives     <- directiveRepository.getFullDirectiveLibrary()
+            } yield {
               val id = hundredReports.headOption match {
                 // None means this is a new rudder without any reports, don't log anything, current id is 0
                 case None =>
                   logger.warn("There is no reports to log")
                   0
 
-                  // return last id and report the latest 100 non compliant reports
+                // return last id and report the latest 100 non compliant reports
                 case Some(report) =>
-                  logReports(hundredReports.map(_._1).reverse)
-                  report._2
+                  logReports(hundredReports.reverse, nodes, rules.map(r => (r.id, r)).toMap, directives)
+                  report._1
               }
               updateLastId(id)
+            }
 
-            case eb:EmptyBox =>
-              logger.error("report logger could not fetch latest 100 non compliant reports, retry on next run",eb)
-          }
-
-        case Full(last) =>
-          logger.trace("***** get current log id")
-
-          reportsRepository.getHighestId match {
-            case Full(currentId) if (currentId > last) =>
-              logger.trace("***** log report beetween ids %d and %d".format(last,currentId))
-              reportsRepository.getErrorReportsBeetween(last,currentId,Reportskind) match {
-                case Full(reports) =>
-                  logReports(reports)
-                  updateLastId(currentId)
-                case eb:EmptyBox =>
-                  logger.error("report logger could not fetch latest non compliant reports, try on next run",eb)
-              }
+            isSuccess match {
+              case eb:EmptyBox =>
+                logger.error("report logger could not fetch latest 100 non compliant reports, retry on next run",eb)
+              case Full(x) => //
+            }
 
-            case _ =>
-              logger.trace("***** no reports to log")
-          }
 
-        case Failure(message,_,_) =>
-          logger.error("could not fetch last id, don't log anything, wait next run, cause is %s".format(message))
-      }
+          case Full(lastId) =>
+            logger.trace("***** get current highest non-compliant log id")
+            val highest = reportsRepository.getHighestErrorId(reportskind)
+            logger.trace(s"***** highest non-compliant log id = ${highest} and last processed id = ${lastId}")
+            highest match {
+              case Full(currentId) if (currentId > lastId) =>
+                logReportsBetween(lastId, currentId)
```

Comment:
I do understand that, but the actual request is: 
"give me 10k the non-compliant reports after 8042 [last processed id] (and not after 8042, because it's the highest non-compliant id). So the actual batch size in your case is 0. And postgres knows there is no repaired/error with the constraint due to the index. 

And if 1000000000045 DOES be a non-compliant report, that does change much: you get one report, update the index, and be done. 

Again, I do understand what you are saying, it would be better to look to non-compliant since a lower time, but that does not seems really bad as it is, or at least I don't understand why you insist at thinking that I'm going to process other reports than non-compliant one again and again. 
The fact that we don't look in the database for the last id seems much worse than that, in comparison.